### PR TITLE
chore: build error with conda-forge's c stdlib dep

### PR DESCRIPTION
**Pull request by s22chan**

### CODE REVIEW

File: .github/workflows/run-tests.yaml [hunk 1]
```diff
@@ -3,10 +3,6 @@ run-name: run unit tests
 on:
   pull_request:
     branches-ignore: ["release-please*"]
-    paths:
```
Comment (s22chan) [inline .github/workflows/run-tests.yaml:6]: why do we have this? it interferes with some stuff (like if someone changed the setup.py build script or docs)

Comment (ovesh) [reply .github/workflows/run-tests.yaml:6]: I figured that we don't want to run unit tests if someone only changes some documentation etc. We can add more paths here, or switch to paths-ignore

File: conda-recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
 python:
-  - 3.8
-  - 3.9
-  - 3.10
-CONDA_BUILD_SYSROOT:
-  - /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk  # [osx]
\ No newline at end of file
+- '3.8'
+- '3.9'
+- '3.10'
```
Comment (ovesh) [inline conda-recipe/conda_build_config.yaml:12]: Is this related to the fix, or just a driveby?

Comment (s22chan) [reply conda-recipe/conda_build_config.yaml:12]: just to make things more uniform

File: .github/actions/build-gk/action.yaml [hunk 3]
```diff
@@ -31,15 +31,15 @@ runs:
...
+        installer-url: https://micro.mamba.pm/api/micromamba/${{ inputs.platform }}/1.5.8
 
     - name: create build env
       shell: bash -l -e {0}
       run: |
         set -x
         if [ ! -d "${HOME}/micromamba/envs/build" ]; then
           micromamba create -yqn build -c conda-forge \
-            boa==0.16.0 mamba==1.5.7 conda==24.1.2 \
+            boa==0.17.0 mamba==1.5.8 conda=24.5 \
```
Comment (ovesh) [inline .github/actions/build-gk/action.yaml:42]: weird, I remember trying the same combination (but removed mamba) and it not working.

**Review by ovesh — APPROVED** [review]
